### PR TITLE
fix build on 64 bit, add locale support and about window inprovements

### PR DESCRIPTION
**Pull request by hfsfox**

### CODE REVIEW

File: source/App.cpp [hunk 1]
```diff
@@ -154,15 +160,13 @@ void App::MessageReceived(BMessage *message) {
...
-		"Use a utility like PatchBay to connect it to your MIDI applications!\n\n"
-		"InternalMIDI is released under MIT license.\n\n"
-		"Project Homepage:\nhttp://dev.osdrawer.net/projects/internalmidi\n", "OK", NULL, NULL, 
-		B_WIDTH_FROM_WIDEST, B_EVEN_SPACING, B_INFO_ALERT))->Go();
+	BAboutWindow* about = new BAboutWindow("InternalMIDI", APP_SIGNATURE);
+	about->AddCopyright(2023,"Werner Freytag");
+	about->AddDescription(B_TRANSLATE("InternalMIDI creates a MIDI node for the internal Haiku General MIDI synthesizer.\n"
+	"Use a utility like PatchBay to connect it to your MIDI applications!\n\n"
+	"InternalMIDI is released under MIT license.\n\n"
+	"Project Homepage:\nhttp://dev.osdrawer.net/projects/internalmidi\n"));
```
Comment (humdingerb) [inline source/App.cpp]: Since the homepage doesn't exist any more, we can as well remove it.

File: source/App.cpp [hunk 2]
```diff
@@ -154,15 +160,13 @@ void App::MessageReceived(BMessage *message) {
...
-	(new BAlert("About InternalMIDI", 
-		"\nInternalMIDI V2.5.3\n\n"
-		"Copyright ©2001 by Werner Freytag\n\n"
-//		"Copyright ©2009/2010 Haiku Inc. \n\n"
-		"InternalMIDI creates a MIDI node for the internal Haiku General MIDI synthesizer.\n"
-		"Use a utility like PatchBay to connect it to your MIDI applications!\n\n"
-		"InternalMIDI is released under MIT license.\n\n"
-		"Project Homepage:\nhttp://dev.osdrawer.net/projects/internalmidi\n", "OK", NULL, NULL, 
-		B_WIDTH_FROM_WIDEST, B_EVEN_SPACING, B_INFO_ALERT))->Go();
+	BAboutWindow* about = new BAboutWindow("InternalMIDI", APP_SIGNATURE);
```
Comment (humdingerb) [inline source/App.cpp]:
```suggestion
	BAboutWindow* about = new BAboutWindow(B_TRANSLATE_SYSTEM_NAME("InternalMIDI"), APP_SIGNATURE);
```

...plus anywhere the app name appears inthe interface,

File: source/DeskbarView.cpp [hunk 3]
```diff
@@ -145,22 +149,22 @@ void DeskbarView::PopUpMenu() {
 		
 		BMenuItem	*item;
 	
-		item = new BMenuItem("Settings...", new BMessage('Sett'));
+		item = new BMenuItem(B_TRANSLATE("Settings..."), new BMessage('Sett'));
```
Comment (humdingerb) [inline source/DeskbarView.cpp]:
```suggestion
		item = new BMenuItem(B_TRANSLATE("Settings" B_UTF8_ELLIPSIS), new BMessage('Sett'));
```

File: source/DeskbarView.cpp [hunk 4]
```diff
@@ -145,22 +149,22 @@ void DeskbarView::PopUpMenu() {
...
 		fPopUpMenu->AddItem( item );
 
-		item = new BMenuItem("Show this replicant", new BMessage('Repl'));
+		item = new BMenuItem(B_TRANSLATE("Show this replicant"), new BMessage('Repl'));
 		item->SetMarked( true );
 		item->SetTarget( messenger );
 		fPopUpMenu->AddItem( item );
 
-		item = new BMenuItem("About...", new BMessage( B_ABOUT_REQUESTED ));
+		item = new BMenuItem(B_TRANSLATE("About..."), new BMessage( B_ABOUT_REQUESTED ));
```
Comment (humdingerb) [inline source/DeskbarView.cpp]:
```suggestion
		item = new BMenuItem(B_TRANSLATE("About InternalMIDI"), new BMessage( B_ABOUT_REQUESTED));
```

**Review by humdingerb — COMMENTED** [review]
The interesting part comes when doing the layout management & localization for the settings window... :)

**Review by humdingerb — COMMENTED** [review]
When introducing localization, it's wise to using Haiku's layout management as well to account for labels etc. being of different sizes for different languages.
That means, using the layout-aware constructors for all views and widgets etc. Mostly, that's a constructor without the BRect frame parameter. For a BView, for example, see the [Haiku Book](https://www.haiku-os.org/docs/api/layout_intro.html). To arrange the widgets, it's best to use [BLayoutBuilder](https://www.haiku-os.org/docs/api/LayoutBuilder_8h.html)

**Comment by Begasus** [comment]
@humdingerb what's the status on this one?

**Comment by humdingerb** [comment]
Sorry, I missed @hfsfox changes... Merging.

**Pull request merged**